### PR TITLE
feat(plasma): send perftool report

### PR DESCRIPTION
**Pull request by KateKate**
:baby_chick: Download canary assets:

[colors--canary.323.4145566448.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.323.4145566448.xml)  
[color_metro_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.323.4145566448.swift)  
[color_metro_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.323.4145566448.kt)  
[color_metro_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.323.4145566448.ts)  
[color_sbermarket_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.323.4145566448.swift)  
[color_sbermarket_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.323.4145566448.kt)  
[color_sbermarket_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.323.4145566448.ts)  
[color_sberprime_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.323.4145566448.swift)  
[color_sberprime_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.323.4145566448.kt)  
[color_sberprime_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.323.4145566448.ts)  
[color_selgros_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.323.4145566448.swift)  
[color_selgros_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.323.4145566448.kt)  
[color_selgros_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.323.4145566448.ts)  
[color_smbusiness_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.323.4145566448.swift)  
[color_smbusiness_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.323.4145566448.kt)  
[color_smbusiness_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.323.4145566448.ts)  
[PlasmaTokensColor--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.323.4145566448.swift)  
[shadow_sbermarket_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.323.4145566448.ts)  
[typo_mage_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.323.4145566448.swift)  
[typo_mage_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.323.4145566448.kt)  
[typo_mage_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.323.4145566448.ts)  
[typo_plasma_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.323.4145566448.swift)  
[typo_plasma_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.323.4145566448.kt)  
[typo_plasma_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.323.4145566448.ts)  
[typo_ruler_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.323.4145566448.swift)  
[typo_ruler_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.323.4145566448.kt)  
[typo_ruler_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.323.4145566448.ts)  
[typo_sage_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.323.4145566448.swift)  
[typo_sage_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.323.4145566448.kt)  
[typo_sage_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.323.4145566448.ts)  
[typo_sbermarket_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.323.4145566448.swift)  
[typo_sbermarket_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.323.4145566448.kt)  
[typo_sbermarket_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.323.4145566448.ts)  
[typo_soulmate_ios-swift--canary.323.4145566448.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.323.4145566448.swift)  
[typo_soulmate_kotlin--canary.323.4145566448.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.323.4145566448.kt)  
[typo_soulmate_react-native--canary.323.4145566448.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.323.4145566448.ts)

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

File: scripts/perftool-send-report.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
```
[the remark this answers is not in the document]
Comment (Yeti-or) [reply scripts/perftool-send-report.js:1]: зачем?

File: .github/workflows/performance-test-pr.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ jobs:
       - name: Setup packages
         run: |
           npm ci
-          npx lerna bootstrap
```
Comment (Yeti-or) [inline .github/workflows/performance-test-pr.yml:38]: это же не будет работать если есть изменения в core
давайте `--include-dependencies` добавим?

File: package.json [hunk 3]
```diff
@@ -134,4 +135,4 @@
   "dependencies": {
     "babel-plugin-annotate-pure-calls": "0.4.0"
   }
-}
```
Comment (Yeti-or) [inline package.json:137]: кажется редактор неверно настроен? или prettier =/

File: scripts/perftool-send-report.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+/* eslint-disable no-continue */
+
+/* eslint-disable no-console,  @typescript-eslint/no-var-requires */
+const path = require('path');
+const uuid = require('uuid');
```
Comment (Yeti-or) [inline scripts/perftool-send-report.js:6]: можем обновиться кстати на v14.17.x и тогда можем из crypto забрать этот метод

File: scripts/perftool-send-report.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+const fetch = require('node-fetch');
```
Comment (Yeti-or) [inline scripts/perftool-send-report.js:7]: в любом случае не ясно откуда берутся эти зависимости, ну точнее я догадываюсь что где-то из недр node_modules в каком-то случайном пакете они есть, но это не безопасно, надо их прописать в package.json

File: scripts/perftool-send-report.js (hunk 4 above)
Comment (Yeti-or) [reply scripts/perftool-send-report.js:6]: https://nodejs.org/docs/latest-v14.x/api/crypto.html#crypto_crypto_randomuuid_options

File: .github/workflows/performance-test-pr.yml (hunk 2 above)
Comment (akhdrv) [reply .github/workflows/performance-test-pr.yml:38]: замержим https://github.com/salute-developers/plasma/pull/328 и можно убрать скоуп

Comment (KateKate) [reply .github/workflows/performance-test-pr.yml:38]: @akhdrv замержи 328, там я тоже лайкнула. и давай тут ребейзну и уберу --scope "@salutejs/plasma-ui"

File: scripts/perftool-send-report.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+const URL = 'https://metrics.prom.third-party-app.sberdevices.ru/create-canvas-app-metrics';
+
+async function perftoolSendReport() {
+    const { staticTaskChange, result } = await fsPromises
+        .readFile(path.resolve('./packages/plasma-ui/perftest/comparison.json'), { encoding: 'utf-8' })
+        .then(JSON.parse);
+
+    // TODO: обсудить с Артемом нужно ли это все-таки
+    // if (!isVersionChanged) {
```
[the remark this answers is not in the document]
Comment (KateKate) [reply scripts/perftool-send-report.js:39]: а расскажи когда isVersionChanged = true? вот что нашла
```
const { version: currentVersion, result: currentResult, staticTaskResult: currentStaticTaskResult } = currentReport;
    const {
        version: previousVersion,
        result: previousResult,
        staticTaskResult: previousStaticTaskResult,
    } = previousReport;

    const stabilizers = [];
    const isVersionChanged = currentVersion !== previousVersion;
    if (isVersionChanged) {
        warn('Looks like perftool version is changed. Some results may be falsy');
    }
```

выглядит так что это изменение версии самой либы - perftool, но нам же это не подходит

File: scripts/perftool-send-report.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (KateKate) [reply scripts/perftool-send-report.js:1]: это же в js надо будет транспайлить как-то + тут просто достаточно, давайте js оставим

File: scripts/perftool-send-report.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                        body.push(item);
+                    }
+                }
+            }
+        }
+    } catch (error) {
+        console.error('Body creation Error', error);
+    }
+
+    const message = 'Sent report';
```
[the remark this answers is not in the document]
Comment (KateKate) [reply scripts/perftool-send-report.js:79]: да просто - так как в двух местах, могу заинлайнить

File: scripts/perftool-send-report.js (hunk 4 above)
Comment (KateKate) [reply scripts/perftool-send-report.js:6]: а у нас же там `Use Node.js 16.15.x`. Спасибо, заюзаю crypto

File: scripts/perftool-send-report.js (hunk 5 above)
Comment (KateKate) [reply scripts/perftool-send-report.js:7]: оно уже там есть https://github.com/salute-developers/plasma/blob/master/package.json#L95

File: scripts/perftool-send-report.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (akhdrv) [reply scripts/perftool-send-report.js:39]: просто в ПР где версия изменилась, данные сравнения могут быть невалидные и их не надо рисовать на дашбордах

File: .github/workflows/performance-test-pr.yml (hunk 2 above)
Comment (akhdrv) [reply .github/workflows/performance-test-pr.yml:38]: чет там не все хорошо работает. разберусь и замержу

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

File: .github/workflows/performance-test-pr.yml (hunk 2 above)
Comment (KateKate) [reply .github/workflows/performance-test-pr.yml:38]: Ребята, я вроде все поправила, что комментили. Спасибо вам за ревью! Давайте вмержим, чтобы статистику собирать. Пока предлагаю оставить `--scope "@salutejs/plasma-ui"`, где оно будет падать, то пока игнорируем.

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-323/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-323/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-323/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-323/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-323/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-323/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-323/temple-storybook/

**Pull request merged**